exp/api/v1beta1/azuremanagedcluster_webhook.go

```diff
@@ -51,6 +52,21 @@ func (r *AzureManagedCluster) ValidateCreate() error {
 			"can be set only if the AKS feature flag is enabled",
 		)
 	}
+	if !r.Status.Ready {
```

Comment:
Is there a reason we're only checking the status on creates but not updates?

Reply:
Naively I was thinking to protect the "user accidentally includes a value in the `AzureManagedCluster` resource spec when creating the cluster, but not including this check in update leaves open the edge case of "user updates this property after cluster create but before the AKS API does *its* update", and so yeah, I'll add it there as well.

Reply:
In addition to your thought process, would it be worth it to hide `AzureManagedCluster.Spec.ControlPlaneEndpoint` in the CRD from the user ? 
It would [technically be part of the spec](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html#infracluster-resources), just not visible to the user. Could add a psychological barrier on updating a non-exposed field.

Reply:
That sounds like a breaking API change, no?

Reply:
> hide AzureManagedCluster.Spec.ControlPlaneEndpoint in the CRD from the user

Hmm, yeah, that would be a breaking change. We can ignore my comment https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2711#discussion_r1046525392